includes/embeds/class-amp-gallery-embed.php

```diff
@@ -289,4 +291,19 @@ public function print_styles() {
 		</style>
 		<?php
 	}
+
+	/**
+	 * Add lightbox attribute to a provided attachment.
+	 *
+	 * @param array        $attr       Attributes for the image markup.
+	 * @param WP_Post      $attachment Image attachment post.
+	 * @param string|array $size       Requested size. Image size or array of
+	 *                                 width and height values (in that order).
+	 *                                 Default 'thumbnail'.
+	 * @return array Modified attributes for the image markup.
+	 */
+	public function add_lightbox_attribute( $attr, $attachment, $size ) {
```

Comment:
Do you think this would be better as a closure? The above could then be:

```php
$add_lightbox_attribute = static function( $attr ) {
		$attr['lightbox'] = '';
		return $attr;
};
add_filter( 'wp_get_attachment_image_attributes', $add_lightbox_attribute );
$attributes['link'] = 'none';
$html               = gallery_shortcode( $attributes );
remove_filter( 'wp_get_attachment_image_attributes', $add_lightbox_attribute );
```

This would avoid adding to the public API space.

Not a big deal, however.

Also, the `$attachment` and `$size` attributes are not used so they could be left off.

Reply:
Solved in https://github.com/ampproject/amp-wp/pull/3097/commits/a9e18f9cd0f1a7a37977400f964933b3b7a6cfe0